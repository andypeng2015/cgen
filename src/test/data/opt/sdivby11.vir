module foo

export function w $foo(w %x) {
@start:
  %x = div.w %x, 11_w
  ret %x
}
