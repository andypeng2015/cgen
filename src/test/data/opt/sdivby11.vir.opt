module foo

export function w $foo(w %x) {
@2:
  %3 = mulh.w %x, 0x1745d175_w ; @7
  %4 = lsr.w %3, 0x1f_w ; @8
  %5 = add.w %3, %4 ; @9
  ret %5
}
