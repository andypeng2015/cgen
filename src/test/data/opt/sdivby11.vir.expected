module foo

export function w $foo(w %x) {
@2:
  %0 = mulh.w %x, 0x1745d175_w ; @4
  %1 = lsr.w %0, 0x1f_w ; @5
  %2 = add.w %0, %1 ; @6
  ret %2
}
